lighthouse-core/lib/page-functions.js

```diff
@@ -155,15 +155,33 @@ function getOuterHTMLSnippet(element, ignoreAttrs = [], snippetCharacterLimit =
     for (const attributeName of clone.getAttributeNames()) {
       if (charCount > snippetCharacterLimit) {
         clone.removeAttribute(attributeName);
-      } else {
-        let attributeValue = clone.getAttribute(attributeName);
-        if (attributeValue === null) continue;
-        if (attributeValue.length > ATTRIBUTE_CHAR_LIMIT) {
-          attributeValue = attributeValue.slice(0, ATTRIBUTE_CHAR_LIMIT - 1) + '…';
-          clone.setAttribute(attributeName, attributeValue);
+        continue;
+      }
+
+      let attributeValue = clone.getAttribute(attributeName);
+      if (attributeValue === null) continue; // Can't happen.
+
+      let dirty = false;
+
+      // Replace img.src with img.currentSrc.
+      if (attributeName === 'src' && element.tagName === 'IMG') {
+        const currentSrc = /** @type {HTMLImageElement} */ (element).currentSrc;
+        // Only replace if the two URLs do not resolve to the same location.
+        const documentHref = element.ownerDocument.location.href;
+        if (new URL(attributeValue, documentHref).toString() !== currentSrc) {
```

Comment:
my initial instinct was that this will throw for invalid values, but I recently learned ( #12008) that Chrome's URL is waaaaay more lenient than node's. Worth a test/catch anyway?

The failure mode for `getOuterHTMLSnippet` is quite painful in that it will typically fail the entire gatherer

Reply:
This entire block is wrapped in a try catch, so the failure mode is just that the snippet is `<tagName>`.

Want me to pull this out to another function w/ another try/catch?

Reply:
oh I totally missed that I was chasing the chain upwards but it was right here elided 😆 

sorry for the noise!